File: Playwright1/tests/3 DragNDrop.test.js
import { test, expect } from '@playwright/test'

test('drag and drop test', async ({ page }) => {
    await page.goto("https://jqueryui.com/droppable/");
    console.log("Page opened successfully");

    //Find the source, target elements to drag and drop
    await page.waitForTimeout(2000);

    const sourceElement = await page.frameLocator('.demo-frame').getByText("Drag me to my target");
    const targetElement = await page.frameLocator('.demo-frame').getByText("Drop here");
    
    await sourceElement.dragTo(targetElement);
    await page.waitForTimeout(3000);
})

test('drag and drop test2', async ({ page }) => {
    await page.goto("https://jqueryui.com/droppable/");
    console.log("Page opened successfully");

    //Find the source, target elements to drag and drop
    await page.waitForTimeout(2000);

    const frame = page.frameLocator("//iframe[@src='/resources/demos/droppable/default.html']");

    const sourceElement = await frame.getByText("Drag me to my target");
    const targetElement = await frame.getByText("Drop here");
    
    await sourceElement.dragTo(targetElement);
    await page.waitForTimeout(3000);
})

File: Playwright_CS/PlaywrightTests/3 DragNDropTest.cs
using System.Text.RegularExpressions;
using System.Threading.Tasks;
using Microsoft.Playwright;
using Microsoft.Playwright.NUnit;
using NUnit.Framework;

namespace PlaywrightTests;

[Parallelizable(ParallelScope.Self)]
[TestFixture]
public class DragAndDropTest : PageTest
{
    [Test]
    public async Task DragNDropTest()
    {
        Console.WriteLine("Executing DragNDropTest");
        await Page.GotoAsync("https://jqueryui.com/droppable/");
        await Page.WaitForTimeoutAsync(3000);

        ILocator sourceElement = Page.FrameLocator(".demo-frame").GetByText("Drag me to my target");
        ILocator targetElement = Page.FrameLocator(".demo-frame").GetByText("Drop here");

        await sourceElement.DragToAsync(targetElement);
        await Page.WaitForTimeoutAsync(3000);
    }

    // [Test]
    // public async Task DragNDropTes2()
    // {
    //     Console.WriteLine("Executing DragNDropTest2");
    //     await Page.GotoAsync("https://jqueryui.com/droppable/");
    //     await Page.WaitForTimeoutAsync(3000);

    //     var frameElementHandle = await Page.EvaluateAsync("window.frames[0]");
    //     var frame = frameElementHandle

    //     ILocator sourceElement = frameElement.GetByText("Drag me to my target");
    //     ILocator targetElement = Page.FrameLocator(".demo-frame").GetByText("Drop here");

    //     await sourceElement.DragToAsync(targetElement);
    //     await Page.WaitForTimeoutAsync(3000);
    // }
}